src/vmm/src/config.rs

```diff
@@ -48,23 +68,16 @@ impl TryFrom<String> for MemoryConfig {
 
     fn try_from(mem_cfg_str: String) -> result::Result<Self, Self::Error> {
         // Supported options: `size=<u32>`
-        let mem_cfg_str_lower = mem_cfg_str.to_lowercase();
-        let tokens: Vec<&str> = mem_cfg_str_lower
-            .split('=')
-            .filter(|tok| !tok.is_empty())
-            .collect();
-        if tokens.len() != 2 {
-            return Err(ConversionError::ParseMemory(mem_cfg_str));
-        }
-        if tokens[0] != "size_mib" {
-            return Err(ConversionError::ParseMemory(tokens[0].to_string()));
-        }
-        tokens[1]
-            .parse::<u32>()
-            .map(|mem_size_mib| MemoryConfig {
-                size_mib: mem_size_mib,
-            })
-            .map_err(|_| ConversionError::ParseMemory(tokens[1].to_string()))
+        let mut arg_parser = CfgArgParser::new(mem_cfg_str);
```

Comment:
This looks a lot cleaner now! The new struct and associated functions are quite clean right?